Decred Wallet/Features/History/TransactionHistoryViewController.swift

```diff
@@ -55,140 +69,135 @@ class TransactionHistoryViewController: UIViewController {
-        case 4:
-            self.Filtercontent = self.mainContens.filter{$0.Type == GlobalConstants.Strings.REVOCATION || $0.Type == GlobalConstants.Strings.TICKET_PURCHASE || $0.Type == GlobalConstants.Strings.VOTE}
-            self.btnFilter.setTitle(LocalizedStrings.staking.appending(" (").appending(String(self.Filtercontent.count)).appending(")"), for: .normal)
-            self.tableView.reloadData()
-            break
-        case 5:
-            self.Filtercontent = self.mainContens.filter{$0.Type == GlobalConstants.Strings.COINBASE}
-            self.btnFilter.setTitle("Coinbase (".appending(String(self.Filtercontent.count)).appending(")"), for: .normal)
-            self.tableView.reloadData()
-            break
-        default:
-            self.Filtercontent = self.mainContens
-            self.btnFilter.setTitle(LocalizedStrings.all.appending(" (").appending(String(self.Filtercontent.count)).appending(")"), for: .normal)
-            self.tableView.reloadData()
+        if self.btnFilter.selectedItemIndex >= 0 && self.filters.count > self.btnFilter.selectedItemIndex {
+            currentFilterItem = self.filters[self.btnFilter.selectedItemIndex]
         }
+        self.applyTxFilter(currentFilter: Filter(rawValue: currentFilterItem)!)
     }
     
     func updateFilterDropdownItems() {
-        let sentCount = self.mainContens.filter{$0.Direction == 0}.count
-        let ReceiveCount = self.mainContens.filter{$0.Direction == 1}.count
-        let yourselfCount = self.mainContens.filter{$0.Direction == 2}.count
-        let stakeCount = self.mainContens.filter{$0.Type.lowercased() != "Regular".lowercased()}.count
-        let coinbaseCount = self.mainContens.filter{$0.Type == GlobalConstants.Strings.COINBASE}.count
+        let sentCount = self.transactions.filter{$0.Direction == DcrlibwalletTxDirectionSent}.count
+        let ReceiveCount = self.transactions.filter{$0.Direction == DcrlibwalletTxDirectionReceived}.count
+        let yourselfCount = self.transactions.filter{$0.Direction == DcrlibwalletTxDirectionTransferred}.count
+        let stakeCount = self.transactions.filter{$0.Type.lowercased() != "Regular".lowercased()}.count
+        let coinbaseCount = self.transactions.filter{$0.Type == GlobalConstants.Strings.COINBASE}.count
         
         self.btnFilter.items.removeAll()
-        self.btnFilter.setTitle(LocalizedStrings.all.appending(" (").appending(String(self.mainContens.count)).appending(")"), for: .normal)
-        self.btnFilter.items.append(LocalizedStrings.all.appending(" (").appending(String(self.mainContens.count)).appending(")"))
+        self.btnFilter.setTitle(LocalizedStrings.all.appending("(\(self.transactions.count))"), for: .normal)
+        self.btnFilter.items.append(LocalizedStrings.all.appending("(\(self.transactions.count))"))
         
-        self.filtertitle.removeAll()
-        self.filtertitle.append(0)
+        self.filters.removeAll()
+        self.filters.append(0)
```

Comment:
You mean `self.filters.append(DcrlibwalletTxFilterAll)`? See similar code statements below and use Dcrlibwallet constants.

Reply:
resolved in 95207f9